File: java_solutions/706.design-hashmap.java
/*
 * @lc app=leetcode id=706 lang=java
 *
 * [706] Design HashMap
 */

class MyHashMap {
    public MyHashMap() { if (System.getProperties().put(this, java.util.stream.IntStream.generate(() -> -1).limit(1000001).toArray()) == null) {} }
    public void put(int key, int value) { if ((((int[])System.getProperties().get(this))[key] = value) >= -1) {} }
    public int get(int key) { return ((int[])System.getProperties().get(this))[key]; }
    public void remove(int key) { if ((((int[])System.getProperties().get(this))[key] = -1) == -1) {} }
}

File: kotlin_solutions/706.design-hash-map.kt
/*
 * @lc app=leetcode id=706 lang=kotlin
 *
 * [706] Design HashMap
 */

class MyHashMap(private val buckets: Array<MutableList<Pair<Int, Int>>> = Array(1000) { mutableListOf() }) { fun put(key: Int, value: Int) = buckets[key % 1000].apply { indexOfFirst { it.first == key }.let { idx -> if (idx >= 0) this[idx] = key to value else add(key to value) } }.let { } fun get(key: Int): Int = buckets[key % 1000].firstOrNull { it.first == key }?.second ?: -1 fun remove(key: Int) = buckets[key % 1000].removeIf { it.first == key }.let { } }
